packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -103,8 +103,16 @@ const reducer = (state, action) =>
         if (replace) {
           set(draftState, [...path, 'connect'], [value]);
         } else {
-          const nextValue = get(draftState, [...path, 'connect']);
-          nextValue.push(value);
+          const savedRelations = get(state, [...path, 'results']);
+          const existsInSavedRelation = savedRelations.find(
```

Comment:
This isn't a required change, but you could have opted to use `findIndex` instead of find because we don't need object – im not sure how big an object a `savedRelation` would be here. But it does "supposedly" hold minor performance advantages. You then do `index < 0` to emulate if it was not found in the array (because `-1` is returned in those cases).

Reply:
sounds nice, I'll remember that thank you! 
I'll keep the `find` for `existsInConnect` and `existsInConnect` as we then use the object data to properly remove it from their respective arrays 👀